source/shared/cpp/ObjectModel/ParseUtil.cpp

```diff
@@ -403,6 +403,23 @@ std::vector<std::shared_ptr<BaseActionElement>> ParseUtil::GetActionCollection(
     return elements;
 }
 
+std::shared_ptr<BaseActionElement> ParseUtil::GetSelectAction(
+    std::shared_ptr<ElementParserRegistration> elementParserRegistration,
+    std::shared_ptr<ActionParserRegistration> actionParserRegistration,
+    const Json::Value& json,
+    AdaptiveCardSchemaKey key,
+    bool isRequired)
+{
+    auto selectAction = ParseUtil::ExtractJsonValue(json, key, isRequired);
+
+    if (!selectAction.empty())
+    {
+        return ParseUtil::GetActionFromJsonValue(elementParserRegistration, actionParserRegistration, selectAction);
+    }
+
+    return nullptr;
+}
```

Comment:
Since you've added this, can you call it from BaseCardElement::DeserializeSelectAction? Or alternatively, have the callers of BaseCardElement::DeserializeSelectAction call this instead and remove BaseCardElement::DeserializeSelectAction entirely.

Reply:
updated in b8f2c66